lms/lmsweb/views.py

```diff
@@ -236,6 +236,68 @@ def read_all_notification():
     return jsonify({'success': success_state})
 
 
+@webapp.route('/share', methods=['POST'])
+@login_required
+def share():
+    solution_id = int(request.json.get('solutionId', 0))
+    solution = Solution.get_or_none(solution_id)
+    if solution is None:
+        return fail(404, f'No such solution {solution_id}')
+
+    solver_id = solution.solver.id
+    if solver_id != current_user.id and not current_user.role.is_manager:
+        return fail(403, "You aren't allowed to access this page.")
+
+    if not SHAREABLE_SOLUTIONS:
+        return fail(404, 'Shareable solutions are not allowed.')
+
+    shared_solution = SharedSolution.get_or_none(
+        SharedSolution.solution == solution,
+    )
+    if shared_solution is None:
+        SharedSolution.create_shared_solution(
+            solution=solution,
+        )
+        return jsonify({
+            'success': 'true',
+            'shared': True,
+        })
+
+    shared_solution.delete_instance()
+
+    return jsonify({
+        'success': 'true',
+        'shared': False,
+    })
+
+
+@webapp.route('/share-link', methods=['POST'])
+@login_required
+def share_link():
+    solution_id = int(request.json.get('solutionId', 0))
+    solution = Solution.get_or_none(solution_id)
+    if solution is None:
+        return fail(404, f'No such solution {solution_id}')
+
+    solver_id = solution.solver.id
+    if solver_id != current_user.id and not current_user.role.is_manager:
+        return fail(403, "You aren't allowed to access this page.")
+
+    if not SHAREABLE_SOLUTIONS:
+        return fail(404, 'Shareable solutions are not allowed.')
+
+    shared_solution = SharedSolution.get_or_none(
+        SharedSolution.solution == solution,
+    )
```

Comment:
Thought about creating a function that will return the shared_solution because the code in this function until this line is the same as the function above (share) until the middle of it.
Would like to hear your opinion.

Reply:
view.py should include as less code as possible. Try to export it to modules/share_link.py